fdbclient/NativeAPI.actor.cpp

```diff
@@ -9759,6 +9828,80 @@ Future<Void> DatabaseContext::waitPurgeGranulesComplete(Key purgeKey) {
 	return waitPurgeGranulesCompleteActor(Reference<DatabaseContext>::addRef(this), purgeKey);
 }
 
+ACTOR Future<bool> setBlobRangeActor(Reference<DatabaseContext> cx, KeyRange range, bool active) {
+	state Database db(cx);
+	state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(db);
+
+	state Value value = active ? blobRangeActive : blobRangeInactive;
+
+	loop {
+		try {
+			tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
+			tr->setOption(FDBTransactionOptions::PRIORITY_SYSTEM_IMMEDIATE);
+
+			if (active) {
+				state RangeResult results = wait(krmGetRanges(tr, blobRangeKeys.begin, range, 5));
```

Comment:
5 seems kind of like an arbitrary constant here.
Also returning false if the range is already set breaks idempotency on retry.
I think what'd we'd want, based on previous discussion with evan, was that this should succeed if exactly the specified range was already blobbified as a single range, but if only part of the specified range was (or it was blobbified as multiple ranges), it should fail

Reply:
Remove the 5 and fixed the failure case.